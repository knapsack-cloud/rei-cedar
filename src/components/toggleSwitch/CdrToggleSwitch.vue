<template>
  <div :class="style['cdr-toggle-switch']">
    <span :id="id" :class="style['cdr-toggle-switch__label']">{{ label }}</span>
    <button @click="(e) => emit('update:modelValue', e)"
 :class="style['cdr-toggle-switch__button']" role="switch" :aria-checked="modelValue" :aria-labelledby="id">
      <span>
        <icon-check-sm :class="style['cdr-toggle-switch__button-icon']" />
      </span>
      <span>
        <icon-x-sm :class="style['cdr-toggle-switch__button-icon']" />
      </span>
      <div :class="style['cdr-toggle-switch__slider']"></div>
    </button>
  </div>
</template>

<script setup>
import { useCssModule, computed } from 'vue';
import IconCheckSm from '../icon/comps/check-sm.vue';
import IconXSm from '../icon/comps/x-sm.vue';
const emit = defineEmits(['update:modelValue'])

const props = defineProps({
  label: {
    type: String
  },
  id: {
    type: String,
    required: true,
  },
  modelValue: {
    type: Boolean,
    default: false
  }
})
const onClick = (e) => {
  props.checked = !props.checked
  emit('switch', e);
}
const style = useCssModule();
const baseClass = 'cdr-toggle-switch';
</script>

<style lang="scss" module src="./styles/CdrToggleSwitch.scss">
</style>
