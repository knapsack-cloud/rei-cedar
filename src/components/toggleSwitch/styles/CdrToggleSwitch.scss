@import '../../../styles/settings/index.scss';
@import './vars/CdrToggleSwitch.vars.scss';

.cdr-toggle-switch {
  @include cdr-toggle-switch-base-mixin;
  display: flex;
  &__label {
    margin: 0 4px;
  }
  &__button {
    position: relative;
    overflow: hidden;
    margin: 0 4px;
    padding: 0;
    border: 1px solid $cdr-color-border-secondary;
    border-radius: 14px;
    gap: 8px;
    width: 43px;
    height: 22px;
    background-color: $cdr-color-background-secondary;

    &:focus {
      border: 1px solid $cdr-color-border-secondary;
      outline: 1px solid $cdr-color-border-secondary;
    }

    &:hover:not(:focus) {
      border: 1px solid $cdr-color-text-button-secondary-hover;
      .cdr-toggle-switch__button-icon {
        fill: $cdr-color-text-button-secondary-hover;
      }
      .cdr-toggle-switch__slider {
        &::before {
          border: 1px solid $cdr-color-text-button-secondary-hover;
        }
      }
    }

    &[aria-checked="true"] {
      background-color: $cdr-color-text-button-secondary;
      border: none;
      
      &:focus {
        border: 1px solid $cdr-color-border-secondary;
        outline: 1px solid $cdr-color-border-secondary;
      }
      &:hover:not(:focus) {
        border: 1px solid $cdr-color-border-secondary;
        .cdr-toggle-switch__button-icon {
          fill: #DCD6CB;
        }
      }
      .cdr-toggle-switch__button-icon {
        fill: #DCD6CB;
      }

      .cdr-toggle-switch__slider {
        &::before {
          left: 22px;
          border: 1px solid $cdr-color-border-secondary;
        }
      }
    }
  }

  &__button-icon {
    height: 17px;
    width: 17px;
  }

  &__slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    -webkit-transition: 0.4s;
    transition: 0.4s;

    &::before {
      position: absolute;
      content: "";
      border: 1px solid $cdr-color-border-secondary;
      border-radius: 14px;
      height: 14px;
      width: 14px;
      top: 3px;
      left: 3px;
      bottom: 4px;
      background-color: #DCD6CB;
      -webkit-transition: left 0.4s, right 0.4s;
      transition: left 0.4s, right 0.4s;
    }
  }
}