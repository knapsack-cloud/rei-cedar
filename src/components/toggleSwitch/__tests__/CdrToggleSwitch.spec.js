import { shallowMount } from '../../../../test/vue-jest-style-workaround.js';
import CdrToggleSwitch from '../CdrToggleSwitch.vue';

describe('CdrToggleSwitch', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(CdrToggleSwitch);
    expect(wrapper.element).toMatchSnapshot();
  });
});
