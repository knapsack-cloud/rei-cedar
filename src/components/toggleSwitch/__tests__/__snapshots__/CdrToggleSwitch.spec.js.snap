// Vitest Snapshot v1

exports[`CdrToggleSwitch > matches snapshot 1`] = `
<div
  class="cdr-toggle-switch"
>
  <span
    class="cdr-toggle-switch__label"
  />
  <button
    aria-checked="false"
    class="cdr-toggle-switch__button"
    role="switch"
  >
    <span>
      <icon-check-sm-stub
        class="cdr-toggle-switch__button-icon"
      />
    </span>
    <span>
      <icon-x-sm-stub
        class="cdr-toggle-switch__button-icon"
      />
    </span>
    <div
      class="cdr-toggle-switch__slider"
    />
  </button>
</div>
`;
